pkg/controller/elasticsearch/driver/nodes.go

```diff
@@ -63,7 +64,13 @@ func (d *defaultDriver) reconcileNodeSpecs(
 		esState:             esState,
 		upscaleStateBuilder: &upscaleStateBuilder{},
 	}
-	if err := HandleUpscaleAndSpecChanges(upscaleCtx, actualStatefulSets, expectedResources); err != nil {
+	actualStatefulSets, err = HandleUpscaleAndSpecChanges(upscaleCtx, actualStatefulSets, expectedResources)
+	if err != nil {
+		return results.WithError(err)
+	}
+
+	// Update PDB to account for new replicas.
+	if err := pdb.Reconcile(d.Client, d.Scheme(), d.ES, actualStatefulSets); err != nil {
```

Comment:
Should this be done after downscaling to account for removed replicas/ssets? Note that HandleUpscaleAndSpecChanges reverts counts to actual if it detects a downscale: https://github.com/elastic/cloud-on-k8s/blob/5258e9015ff578355a305c1add2a838ae4978487/pkg/controller/elasticsearch/driver/upscale.go#L131

Reply:
After downscaling, our StatefulSets express a number of replicas that may be smaller than the number of Pods we currently have running (in the process of being deleted). Building the node count from that is a bit dangerous, we could end up with MinAvailable set to eg. 5 when we have 7 nodes.
I think we want to err on the safe side here: it's OK for MinAvailable to be temporarily higher than it could (if it is set to 8 and we only have 5 nodes right now, then no disruption is allowed, that's no big deal). Computing its value right after upscales is the time where the number of expected nodes should be at its maximum value.

Reply:
We chatted offline as I had some reservations around how long we would stay in a state with too high minAvailable. It seems that this time is not substantial, so this isn't concerning anymore.